libs/gltfio/src/AssetLoader.cpp

```diff
@@ -687,6 +694,46 @@ bool FAssetLoader::createPrimitive(const cgltf_primitive* inPrim, Primitive* out
     return true;
 }
 
+void FAssetLoader::createLight(const cgltf_node* node, Entity entity) {
+    const cgltf_light* light = node->light;
+
+    // Create a name component using the light label if it exists, otherwise use the node label.
+    const char* name = light->name ? light->name : node->name;
+    if (mNameManager && name) {
+        mNameManager->addComponent(entity);
+        mNameManager->setName(mNameManager->getInstance(entity), name);
+    }
+
+    LightManager::Type type = getLightType(light->type);
+    LightManager::Builder builder(type);
+
+    builder.direction({0.0f, 0.0f, -1.0f});
+    builder.color({light->color[0], light->color[1], light->color[2]});
+
+    if (type == LightManager::Type::DIRECTIONAL) {
+        builder.intensity(light->intensity);
+    } else if (type == LightManager::Type::SPOT) {
+        builder.spotLightCone(light->spot_inner_cone_angle, light->spot_outer_cone_angle);
+        // Convert from candelas (luminous intensity) to lumens (luminous power).
+        // lp = li * pi
+        builder.intensity(F_PI * light->intensity);
+    } else if (type == LightManager::Type::POINT) {
+        // Convert from candelas (luminous intensity) to lumens (luminous power).
+        // lp = 4 * pi * li
+        builder.intensity(4.0f * F_PI * light->intensity);
```

Comment:
If it's useful we could also add other intensity() APIs to set the intensity in candelas, like we did for watts.

Reply:
Yeah, I'm not opposed to that. We'd have to name it something different though, like `intensityCandelas`.

Reply:
+1